### PR TITLE
[Enhancement] support count distinct on constant semi-type

### PR DESCRIPTION
**Pull request by Seaven**
## What I'm doing:
image

cast count/sum(distinct xxx) to any_value(xxx)

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctRule.java [hunk 1]
```diff
@@ -54,13 +60,21 @@ public RewriteMultiDistinctRule() {
...
+
         Optional<List<ColumnRefOperator>> distinctCols = Utils.extractCommonDistinctCols(agg.getAggregations().values());
 
         // all distinct function use the same distinct columns, we use the split rule to rewrite
-        return !distinctCols.isPresent();
+        return distinctCols.isEmpty();
     }
 
     public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
+        if (isComplexConstantCountDistinct(input)) {
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctRule.java:75]: How to process `count(distinct row(1, 2), count(distinct col1), count(distinct col2)`?

Comment (Seaven) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/RewriteMultiDistinctRule.java:75]: will rewrite to `any_value(row(1, 2) IS NULL), count(distinct col1), count(distinct col2)`, other will use default strategy

**Review by kangkaisen — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Comment by packy92** [comment]
@mergify backport branch-3.3

**Comment by packy92** [comment]
@mergify backport branch-3.2

**Pull request merged**